### PR TITLE
Upgrade to Java 8 and newest versions of main libraries

### PR DESCRIPTION
**Pull request by pjuza**
**Upgrade to Java 8**

**Issues fromg upgrading to the latest Spring Boot (1.5.1.RELEASE)**
- Spring framework upgrades to v4.3.4.RELEASE
- release notes (https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.5-Release-Notes, https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.4-Release-Notes)
- “ application classes are packaged in BOOT-INF/classes so that the dependent module cannot load a repackaged jar's classes” (http://docs.spring.io/spring-boot/docs/current/maven-plugin/examples/repackage-classifier.html, http://stackoverflow.com/questions/38992527/spring-boot-upgrade-to-1-4-broke-my-multi-module-project-probably-due-to-jar-l)
- “spring-boot.version property was removed from spring-boot-dependencies”
- If you currently listen for an ApplicationStartedEvent in your code you should refactor to use ApplicationStartingEvent. We renamed this class to more accurately reflect what it does.
- “spring.datasource.data” property can contain only valid paths
- @MockBean was introduced in tests
- I had problems to use built-in support for H2 console (see https://openhubframework.atlassian.net/browse/OHFJIRA-24). I decided to leave it in the same state as-is.

**Upgrading Apache Camel to 2.18.2**
- replaced Camel XML configuration by Java configuration (thanks to issue https://issues.apache.org/jira/browse/CAMEL-10109) 

**Upgrading to jaxws-maven-plugin (2.4.1)** - “Failed to read schema document 'xjc.xsd', because 'file' access is not allowed due to restriction set by the accessExternalSchema property" (http://docs.oracle.com/javase/7/docs/api/javax/xml/XMLConstants.html#ACCESS_EXTERNAL_SCHEMA, http://stackoverflow.com/questions/23011547/webservice-client-generation-error-with-jdk8)

**Replaced JODA date/time by JDK8 date/time**
- http://stackoverflow.com/questions/24631909/differences-between-java-8-date-time-api-java-time-and-joda-time, http://blog.joda.org/2014/11/converting-from-joda-time-to-javatime.html
- DateTime -> OffsetDateTime
- created own type for seconds that is often used in configuration - org.openhubframework.openhub.common.time.Seconds
- added dependency to hibernate-java8 (http://stackoverflow.com/questions/39761619/why-is-hibernate-java8-hibernate-5-x-jar-shown-deprecated, http://stackoverflow.com/questions/26447464/persist-java-time-instant-jdk8-with-jpa2-hibernate)

**Side effects:**
- configuration time values can be configured as Duration (not only as Seconds) and then values can be defined according to ISO-8601, e.g.  PT20.345S or PT10H. See java.time.Duration#parse
- we use OffsetDateTime during user interaction (we don’t need to work with zones, therefore we don’t require ZonedDateTime), and Instant in internal (in model objects, in DAO/service layer) use for timestamps 
- date/time values are saved in default time zone into DB
- there is imperfection in admin GUI because there is no information about zone when user wan’t to search messages or logs or something else. We need to add zone information (in HTTP header, directly in date/time string values) to correct it. It’s issue for new admin GUI ...
- JAXB parsing of date/time (org.openhubframework.openhub.api.common.jaxb.JaxbDateAdapter) changed little bit because there were some errors in previous implementation (http://stackoverflow.com/questions/20670041/what-is-the-default-time-zone-for-an-xml-schema-datetime-if-not-specified)

**upgraded main libraries**
Spring Boot: 1.3.4.RELEASE -> 1.5.1.RELEASE
Camel: 2.17.3 -> 2.18.2
JavaMelody: 1.55.0 -> 1.63.0
Jolokia: 1.3.2 -> 1.3.5
IO Metrics: 3.1.0 -> 3.2.0
jaxb2-maven-plugin: 1.5 -> 2.2
upgraded jaxws-maven-plugin: 1.12 -> 2.4.1

Issue: OHFJIRA-1, OHFJIRA-3, OHFJIRA-24, OHFJIRA-18, OHFJIRA-17

### CODE REVIEW

**Comment by hanusto** [comment]
I agree with this PR, but I have problem (and travis also too) with build. See: https://travis-ci.org/OpenWiseSolutions/openhub-framework/builds/208327696.

> [ERROR] Failed to execute goal org.codehaus.mojo:jaxws-maven-plugin:2.4.1:wsimport (WSDL-import-out-asynch-confirmation) on project openhub-core: Not a directory -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:jaxws-maven-plugin:2.4.1:wsimport (WSDL-import-out-asynch-confirmation) on project openhub-core: Not a directory
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Not a directory
        at org.codehaus.mojo.jaxws.WsImportMojo.executeJaxws(WsImportMojo.java:297)
        at org.codehaus.mojo.jaxws.MainWsImportMojo.executeJaxws(MainWsImportMojo.java:51)
        at org.codehaus.mojo.jaxws.AbstractJaxwsMojo.execute(AbstractJaxwsMojo.java:389)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.io.IOException: Not a directory
        at java.io.UnixFileSystem.createFileExclusively(Native Method)
        at java.io.File.createNewFile(File.java:1012)
        at org.codehaus.mojo.jaxws.WsImportMojo.touchStaleFile(WsImportMojo.java:828)
        at org.codehaus.mojo.jaxws.WsImportMojo.processLocalWsdlFiles(WsImportMojo.java:341)
        at org.codehaus.mojo.jaxws.WsImportMojo.executeJaxws(WsImportMojo.java:289)
        ... 24 more

**Comment by pjuza** [comment]
I corrected errors in build => please continue with review.

File: common/src/main/java/org/openhubframework/openhub/common/time/Seconds.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    /**
+     * Creates seconds with specified count.
+     *
+     * @param seconds Count of seconds
+     * @return seconds
+     * @throws IllegalArgumentException if input string cannot be converted to the number
+     */
+    public static Seconds of(String seconds) {
```
Comment (hanusto) [inline common/src/main/java/org/openhubframework/openhub/common/time/Seconds.java:65]: 👍 great, similar as JDK1.8 API

File: core-api/src/main/java/org/openhubframework/openhub/api/common/jaxb/JaxbDateAdapter.java [hunk 2]
```diff
@@ -16,63 +16,91 @@
...
- * JAXB adapter for automatic conversion between {@link XMLGregorianCalendar} to {@link DateTime}.
+ * JAXB adapter for automatic conversion between {@link XMLGregorianCalendar} to {@link OffsetDateTime}.
  *
  * @author Petr Juza
  * @see <a href="http://en.wikipedia.org/wiki/ISO_8601">ISO_8601</a>
  */
 public class JaxbDateAdapter {
 
-    private static final DateTimeFormatter DATE_PRINT_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-ddZZ");
+    // time is set midnight (00:00:00)
```
Comment (hanusto) [inline core-api/src/main/java/org/openhubframework/openhub/api/common/jaxb/JaxbDateAdapter.java:38]: I think that we should have test for it. I did it.

File: web-admin/src/main/resources/sp_h2_server.xml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <beans xmlns="http://www.springframework.org/schema/beans"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd"
-       profile="h2">
+       profile="h2,dev">
```
Comment (hanusto) [inline web-admin/src/main/resources/sp_h2_server.xml:5]: I don`t think so it is correct. OpenHub can run in DEV mode, but for example with PostgreSQL DB.

**Review by hanusto — APPROVED** [review]

**Comment by hanusto** [comment]
From my side it is perfect. In future we should use `Optional` wrapper as NPE restriction. Only one question: in `web-admin/src/main/resources/sp_h2_server.xml` you changed profiles activation. I think that H2 db can run in dev and "prod" mode. Only H2 profile should be required. Please can you explain that?

**Comment by pjuza** [comment]
I have started to use Optional in new changes in the code but I didn't want to refactor the whole code base at once.

sp_h2_server.xml - I think it's vulnerability to allow (even via authentication) web access to DB. I think that only h2 a dev profile should activate web access to DB.

**Comment by hanusto** [comment]
Ok, I agree. It make sense.

**Pull request merged**